feat(copy-trip): add copy trip functionality in the Discovery Page

apana03 opened the pull request:
## Summary

This PR adds the functionality required for copying a trip from the Discover Page to your own trips.

## Changes

- **Models:** added a "copyTrip function in the TripsViewModel.
- **Screens/UI:**  Modified the TripCard composable in the Discover page to include a copy button.


## Checklist

Ensure all / most of these are checked before the pull request is submitted.
- [x] This PR resolves #328.
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached.
- [x] Documentation has been updated.

##  Testing

- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

Added new unit test and also tested the new feature on the emulator.

##  Related Issues/Tickets

Closes #328

## Screenshots (if applicable)

https://github.com/user-attachments/assets/b56a0f7c-bc32-4988-8947-3c9643327360

g0g05arui commented on app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt:
```diff
@@ -437,6 +438,26 @@ open class TripsViewModel(
+   * @param onSuccess A callback invoked upon the successful creation of the copied trip.
+   */
+  fun copyTrip(userViewModel: UserViewModel, onSuccess: () -> Unit) {
+    userViewModel.user
```
This line doesn't seem to do anything

apana03 replied:
Yep, forgot to delete it before pushing, thx for catching it.

g0g05arui commented on app/src/main/java/com/android/voyageur/model/trip/TripsViewModel.kt:
```diff
@@ -437,6 +438,26 @@ open class TripsViewModel(
+  fun copyTrip(userViewModel: UserViewModel, onSuccess: () -> Unit) {
+    userViewModel.user
+    val trip =
+        userViewModel.user.value
```
Here use the id from `Firebase.auth` as all the user data might have not loaded yet as it's an async operation and the `Firebase.auth.uid` is sure to loaded / cached

g0g05arui commented on app/src/main/res/values/strings.xml:
```diff
@@ -30,8 +30,9 @@
     <string name="notification_no_internet_title">No Internet Connection</string>
     <string name="notification_no_internet_text">Please check your network settings.</string>
-
+    <string name="trip_copied_message">%1$s copied successfully!</string>
```
I think this message is not clear enough. From a user's perspective that is new to the application, they press the button, a toast appears and that's it. Maybe it would be better to say something like `Trip copied succesfully, you can find it in the overview screen` or something similar.

g0g05arui reviewed (changes requested):
Great feature! I see you added proper testing for it and we reach over 80% line coverage on this PR and good job adding detailed documentation. 

I just left 2 small requests. After they are ready I will approve.

apana03 commented:
> Great feature! I see you added proper testing for it and we reach over 80% line coverage on this PR and good job adding detailed documentation.
> 
> I just left 2 small requests. After they are ready I will approve.

I addressed your comments. Let me know if it is fine now.
